runtime/compiler/ilgen/Walker.cpp

```diff
@@ -7197,14 +7197,44 @@ TR_J9ByteCodeIlGenerator::storeFlattenableInstance(int32_t cpIndex)
    char * fieldNamePrefix = getTopLevelPrefixForFlattenedFields(owningMethod, cpIndex, prefixLen, comp()->trMemory()->currentStackRegion());
 
    TR_OpaqueClassBlock * containingClass = owningMethod->definingClassFromCPFieldRef(comp(), cpIndex, _methodSymbol->isStatic());
-   const TR::TypeLayout *containingClassLayout = comp()->typeLayout(containingClass);
+   const TR::TypeLayout * containingClassLayout = comp()->typeLayout(containingClass);
    size_t fieldCount = containingClassLayout->count();
 
    TR::Node * value = pop();
    TR::Node * address = pop();
 
+   if (fieldCount == 0)
+      {
+      if (comp()->getOption(TR_TraceILGen))
+         {
+         traceMsg(comp(), "%s: cpIndex %d fieldCount 0 value n%dn isNonNull %d address n%dn isNonNull %d\n", __FUNCTION__, cpIndex,
+            value->getGlobalIndex(), value->isNonNull(), address->getGlobalIndex(), address->isNonNull());
+         }
+
+      // If the field count is greater than zero, the NULLCHKs will be added in the process of
+      // generating IL for the loadInstance and storeInstance below.
+      // If the file count is zero, the NULLCHKs on the references for the value and the target
```

Comment:
Typo:  file -> field

Reply:
Whoops, sorry about it. Address in 60809c4